src/main/java/com/iridium/iridiumskyblock/api/UserChatToggleEvent.java

```diff
@@ -12,10 +12,12 @@
 
     private static final HandlerList handlers = new HandlerList();
     private boolean cancelled;
+    private final boolean isIslandChat;
     private final User user;
 
     public UserChatToggleEvent(User user) {
         this.user = user;
+        this.isIslandChat = user.isIslandChat();
```

Comment:
You need to parse the new state into the event, this is the old state since this event gets called before the state changes